dali/pipeline/data/tensor_list.h

```diff
@@ -68,20 +71,26 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
    * changing the underlying data type if needed.
    */
   template <typename SrcBackend>
-  DLL_PUBLIC inline void Copy(const TensorList<SrcBackend> &other, cudaStream_t stream) {
+  DLL_PUBLIC inline void Copy(const TensorList<SrcBackend> &other, cudaStream_t stream,
+                              bool use_copy_kernel = false) {
     if (IsValidType(other.type())) {
       this->set_type(other.type());
     }
     this->meta_ = other.meta_;
     this->SetLayout(other.GetLayout());
     ResizeLike(other);
-    type_.template Copy<Backend, SrcBackend>(this->raw_mutable_data(),
+
+    if (use_copy_kernel && (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned())) {
+      LaunchCopyKernel(this->raw_mutable_data(), other.raw_data(), this->size() * type_.size(), stream);
+    } else {
+      type_.template Copy<Backend, SrcBackend>(this->raw_mutable_data(),
```

Comment:
Maybe this copy kernel should be hidden inside the `type_.template Copy`?
See dali/pipeline/data/types.cc.